### ISSUE TITLE
Warning 32 unused-value-declaration for functor argument inside a signature

### ISSUE THREAD

**Issue by art-w** [ocaml/ocaml#10721]
There seems to be a problem with the unused value declaration warning in 4.14.0 (but not 4.13.1) :

```ocaml
module type S = sig
  module Make (T : sig val x : int end) : sig end
end
(* File "./test.ml", line 2, characters 23-34:
2 |   module Make (T : sig val x : int end) : sig end
                           ^^^^^^^^^^^
Warning 32 [unused-value-declaration]: unused value x. *)
```
The issue is that the compiler doesn't yet have knowledge of all the implementations of the signature `S`, where `T.x` can actually be required. This pattern can notably be seen in janestreet's base library (for example [comparable_intf.ml](https://github.com/janestreet/base/blob/02baffc343fd0eb8f58dce6fddf1ab5d3afba78e/src/comparable_intf.ml#L155)) which fails to compile with the latest version of trunk 4.14.0+dev0-2021-06-03

**Comment by Octachron** [comment]
Thanks for the report! The issue should be fixed in #10722, and I added your example to the compiler testsuite.

**Comment by art-w** [comment]
That was fast! Thank you!
